forge/csrc/passes/erase_inverse_ops.cpp

```diff
@@ -80,6 +80,23 @@ static std::vector<graphlib::Node *> find_path_to_inverse_op(
     if (not found_inverse)
         path.clear();
 
+    // Skip bypassing the node when binary ops involved in between
```

Comment:
I'm not into introducing this workaround. 

If this takes longer, let's add a feature flag  (`forge/forge/config.py`) and disable erase inverse ops for problematic cases. 

However, let's push as a priority to fix this case properly.